lua/obsidian/init.lua

```diff
@@ -116,12 +116,12 @@ client.vault = function(self)
   local vault_indicator_folder = ".obsidian"
   local dirs = self.dir:parents()
   table.insert(dirs, self.dir:absolute())
-  for _, parent in pairs(self.dir:parents()) do
+  for _, dir in pairs(dirs) do
```

Comment:
This doesn't actually solve anything... `dirs` is equal to `self.dir:parents()`, so nothing is changed by this commit.

Reply:
you need to consider the dir itself, in addition to it's parents. e.g. if the dir is `/a/b/c`, you are only looking at `c`'s parents `/a`, and `/a/b`

Reply:
I'm not too good with lua, but may be something like
```lua
local dirs = { dir, unpack(self.dir:parents()) }
```